securedrop/journalist_app/api.py

```diff
@@ -1,69 +1,69 @@
 import json
 
 from datetime import datetime, timedelta
+from typing import Tuple, Callable, Any
+
+import flask
+import werkzeug
 from flask import abort, Blueprint, current_app, jsonify, request
 from functools import wraps
+
 from sqlalchemy.exc import IntegrityError
 from os import path
 from uuid import UUID
-from werkzeug.exceptions import default_exceptions  # type: ignore
+from werkzeug.exceptions import default_exceptions
 
 from db import db
 from journalist_app import utils
 from models import (Journalist, Reply, Source, Submission,
                     LoginThrottledException, InvalidUsernameException,
                     BadTokenException, WrongPasswordException)
+from sdconfig import SDConfig
 from store import NotEncrypted
 
 
 TOKEN_EXPIRATION_MINS = 60 * 8
 
 
-def get_user_object(request):
-    """Helper function to use in token_required views that need a user
-    object
-    """
-    auth_token = request.headers.get('Authorization').split(" ")[1]
-    user = Journalist.validate_api_token_and_get_user(auth_token)
-    return user
+def _authenticate_user_from_auth_header(request: flask.Request) -> Journalist:
```

Comment:
It would generally be better to make functional changes in separate PRs, particularly when they're optimizations and involve authentication, instead of mixing the refactoring with the type hinting.

I preferred the old name, `get_user_object`: it was accurate, if less precise, and a lot shorter.